content/blog/coredns-in-kubernetes-via-kubeadm.md

```diff
@@ -0,0 +1,268 @@
+[upgrade/staticpods] Waiting for the kubelet to restart the component
+[apiclient] Found 1 Pods for label selector component=kube-controller-manager
+[upgrade/staticpods] Component "kube-controller-manager" upgraded successfully!
+[upgrade/staticpods] Moved new manifest to "/etc/kubernetes/manifests/kube-scheduler.yaml" and backed up old manifest to "/etc/kubernetes/tmp/kubeadm-backup-manifests038673725/kube-scheduler.yaml"
+[upgrade/staticpods] Waiting for the kubelet to restart the component
+[apiclient] Found 1 Pods for label selector component=kube-scheduler
+[upgrade/staticpods] Component "kube-scheduler" upgraded successfully!
+[uploadconfig] Storing the configuration used in ConfigMap "kubeadm-config" in the "kube-system" Namespace
+[bootstraptoken] Configured RBAC rules to allow Node Bootstrap tokens to post CSRs in order for nodes to get long term certificate credentials
+[bootstraptoken] Configured RBAC rules to allow the csrapprover controller automatically approve CSRs from a Node Bootstrap Token
+[bootstraptoken] Configured RBAC rules to allow certificate rotation for all node client certificates in the cluster
+[addons] Applied essential addon: CoreDNS
+[addons] Applied essential addon: kube-proxy
+
+[upgrade/successful] SUCCESS! Your cluster was upgraded to "v1.10.0-alpha.1". Enjoy!
+
+[upgrade/kubelet] Now that your control plane is upgraded, please proceed with upgrading your kubelets in turn.
+
+~~~
+
+## Verifying CoreDNS Service
+To verify CoreDNS is running, we can check the pod status and deployment in the node.
+Note here that the CoreDNS service will remain as "kube-dns" which ensures a smooth transition while upgrading your service discovery from kube-dns to CoreDNS.
+
+Check `pod` status:
+~~~ text
+# kubectl -n kube-system get pods -o wide
+NAME                               READY     STATUS    RESTARTS   AGE       IP              NODE
+coredns-546545bc84-ttsh4           1/1       Running   0          5h        10.32.0.61      sandeep2
+etcd-sandeep2                      1/1       Running   0          5h        147.75.107.43   sandeep2
+kube-apiserver-sandeep2            1/1       Running   0          4h        147.75.107.43   sandeep2
+kube-controller-manager-sandeep2   1/1       Running   0          4h        147.75.107.43   sandeep2
+kube-proxy-fkxmg                   1/1       Running   0          4h        147.75.107.43   sandeep2
+kube-scheduler-sandeep2            1/1       Running   4          5h        147.75.107.43   sandeep2
+weave-net-jhjtc                    2/2       Running   0          5h        147.75.107.43   sandeep2
+~~~
+
+Check `Deployment`:
+~~~ text
+# kubectl -n kube-system get deployments
+NAME      DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+coredns   1            1          1           1        4h
+~~~
+
+We can check if CoreDNS is functioning normally through a few basic `dig` commands:
+
+~~~ text
+# dig @10.32.0.61 kubernetes.default.svc.cluster.local
```

Comment:
make this shorter with `+noall +answer`, noone cares about all that dig metadata